### PR TITLE
Add blog post for kube2pulumi

### PR DESCRIPTION
**Pull request by lblackstone**

### CODE REVIEW

**Comment by lblackstone** [comment]
Preview build is failing because the operator blog post isn't published yet.

File: content/blog/introducing-kube2pulumi/index.md [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+```
+
+`kube2pulumi` supports all of Pulumi's SDKs, so you can choose whatever language you like!
+
+{{< chooser language "typescript,python,csharp,go" >}}
+
+{{% choosable language typescript %}}
+Here, we choose TypeScript and select our YAML manifest.
+
+```
```
Comment (justinvp) [inline content/blog/introducing-kube2pulumi/index.md:46]:
```suggestion
```sh
```

File: content/blog/introducing-kube2pulumi/index.md [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+    },
+});
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+Here, we choose Python and select our YAML manifest.
+
+```
```
Comment (justinvp) [inline content/blog/introducing-kube2pulumi/index.md:69]:
```suggestion
```sh
```

File: content/blog/introducing-kube2pulumi/index.md [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+        "name": "foo",
+    })
+```
+
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+Here, we choose C# and select our YAML manifest.
+
+```
```
Comment (justinvp) [inline content/blog/introducing-kube2pulumi/index.md:91]:
```suggestion
```sh
```

File: content/blog/introducing-kube2pulumi/index.md [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+
+}
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+Here, we choose Go and select our YAML manifest.
+
+```
```
Comment (justinvp) [inline content/blog/introducing-kube2pulumi/index.md:123]:
```suggestion
```sh
```

File: content/blog/introducing-kube2pulumi/index.md [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+Account, Role, RoleBinding, and Deployment, totaling about 130 lines of YAML. Let's try converting that directory
+to some real code!
+
+```
+$ ls operator
+operator.yaml  role.yaml  role_binding.yaml  service_account.yaml
+
+$ kube2pulumi typescript -d operator
+Conversion successful! Generated File: operator/main.ts
+```
```
Comment (justinvp) [inline content/blog/introducing-kube2pulumi/index.md:171]:
````suggestion
```sh
$ ls operator
operator.yaml  role.yaml  role_binding.yaml  service_account.yaml
```

```sh
$ kube2pulumi typescript -d operator
Conversion successful! Generated File: operator/main.ts
```
````

File: content/blog/introducing-kube2pulumi/index.md [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+
+{{% choosable language typescript %}}
+Here, we choose TypeScript and select our YAML manifest.
+
+```
+$ kube2pulumi typescript -f namespace.yaml
+Conversion successful! Generated File: namespace.ts
+```
+
+```ts
```
Comment (justinvp) [inline content/blog/introducing-kube2pulumi/index.md:51]:
```suggestion
```typescript
```

File: content/blog/introducing-kube2pulumi/index.md [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+```
+
+## Superpowers
+
+Now that we've got some Kubernetes resources defined in real code, we can start working on them like any other software
+project. First, we define a variable for the operator name and then swap out hardcoded string references to the other
+resource names. With these changes, it doesn't require any thought to keep these resources in sync! You can later decide
+to change the name in one place, and every other reference will update automatically as well.
+
+```ts
```
Comment (justinvp) [inline content/blog/introducing-kube2pulumi/index.md:355]:
```suggestion
```typescript
```

File: content/blog/introducing-kube2pulumi/index.md [hunk 8]
```diff
@@ -0,0 +1,418 @@
+---
+title: "Introducing kube2pulumi: No More YAML"
+date: 2020-08-14
+draft: true
```
Comment (spara) [inline content/blog/introducing-kube2pulumi/index.md:4]:
```suggestion
```

File: content/blog/introducing-kube2pulumi/index.md [hunk 9]
```diff
@@ -0,0 +1,418 @@
+---
+title: "Introducing kube2pulumi: No More YAML"
+date: 2020-08-14
```
Comment (spara) [inline content/blog/introducing-kube2pulumi/index.md:3]:
```suggestion
date: 2020-08-13
```

File: content/blog/introducing-kube2pulumi/index.md [hunk 10]
```diff
@@ -0,0 +1,416 @@
+---
+title: "Introducing kube2pulumi: No More YAML"
+date: 2020-08-12
```
Comment (spara) [inline content/blog/introducing-kube2pulumi/index.md]:
```suggestion
date: 2020-08-13
```

File: content/blog/introducing-kube2pulumi/index.md [hunk 11]
```diff
@@ -0,0 +1,417 @@
...
+Let's start with a simple example: converting a Namespace. Here's the YAML manifest:
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: foo
+```
+
+`kube2pulumi` supports all of Pulumi's SDKs, so you can choose whatever language you like!
```
Comment (lukehoban) [inline content/blog/introducing-kube2pulumi/index.md:38]: Not sure "Pulumi's SDKs " is clear here - suggestion below - but may need more rewording.

```suggestion
`kube2pulumi` supports all of Pulumi's supported languages, so you can choose whatever language you like!
```

File: content/blog/introducing-kube2pulumi/index.md [hunk 12]
```diff
@@ -0,0 +1,417 @@
...
+supercharge your productivity.
+
+<!--more-->
+
+Using programming languages to manage Kubernetes resources has many benefits: strong typing, linting, auto-completion,
+refactoring, testing, and the ability to use standard language features like variables, loops, conditions, and
+abstraction. Pulumi also supports [Helm], [Kustomize], and [YAML], but these tools don't take full advantage of the
+programming language. With `kube2pulumi`, you can realize these benefits without manual conversion work upfront.
+
+Check out an example of `kube2pulumi` in action, and then we'll take a closer look at the generated code.
```
Comment (lukehoban) [inline content/blog/introducing-kube2pulumi/index.md:25]: Perhaps add a link to and image of pulumi.com/kube2pulumi here?  And then use the post to drill into details?  Now that we have the landing page, it's a great visual and engagement tool to recommend here.

FWIW - we may also want to add a link to that pulumi.com/kube2pulumi page back to this blog post once it's published - "want to learn more about kube2pulumi, check out the blog post for more details".

File: content/blog/introducing-kube2pulumi/index.md [hunk 13]
```diff
@@ -0,0 +1,417 @@
...
+
+That's pretty cool but doesn't show off the full power of `kube2pulumi`. Let's try a more realistic conversion
+scenario. Our newly announced [Pulumi Kubernetes Operator] requires a few different resources to install: a Service
+Account, Role, RoleBinding, and Deployment, totaling about 130 lines of YAML. Let's try converting that directory
+to some real code!
+
+```sh
+$ ls operator
+operator.yaml  role.yaml  role_binding.yaml  service_account.yaml
+```
```
Comment (lukehoban) [inline content/blog/introducing-kube2pulumi/index.md:167]: Do these files exist somewhere we can link to so readers could try this out?  Or could we provide a URL to see the results in pulumi.com/kube2pulumi?

File: content/blog/introducing-kube2pulumi/index.md [hunk 14]
```diff
@@ -0,0 +1,417 @@
...
+
+This simple refactoring is just the tip of the iceberg. At this point, you could:
+
+- wrap these resources in a class to
+abstract away the low-level details,
+- add configuration values to the stack to make it easy to tweak certain values,
+- or even add custom logic for Pulumi to run after a particular resource is created.
+
+Pulumi gives you superpowers, and
```
Comment (lukehoban) [inline content/blog/introducing-kube2pulumi/index.md:391]: Maybe "Kubernetes superpowers"?

File: content/blog/introducing-kube2pulumi/index.md (hunk 13 above)
Comment (viveklak) [reply content/blog/introducing-kube2pulumi/index.md:167]: https://github.com/pulumi/pulumi-kubernetes-operator/tree/master/deploy/yaml
Agree with the above suggestion to crosslink.

File: content/blog/introducing-kube2pulumi/index.md [hunk 15]
```diff
@@ -0,0 +1,417 @@
...
+you decide how to use them!
+
+## Learn More
+
+If you'd like to learn about Pulumi and how to manage your
+infrastructure and Kubernetes through code, [get started today]({{< relref "/docs/get-started" >}}). Pulumi is open
+source and free to use.
+
+For further examples on how to use Pulumi to create Kubernetes
+clusters, or deploy workloads to a cluster, check out the rest of the
```
Comment (viveklak) [inline content/blog/introducing-kube2pulumi/index.md:401]: Nit - Perhaps call out that we are only scratching the surface with these "super powers" and that we welcome feedback and feature requests?

File: content/blog/introducing-kube2pulumi/index.md (hunk 13 above)
Comment (lblackstone) [reply content/blog/introducing-kube2pulumi/index.md:172]: Linked

**Pull request merged**